src/server/v2.0/handler/retention.go

```diff
@@ -391,3 +415,34 @@ func (r *retentionAPI) requirePolicyAccess(ctx context.Context, p *policy.Metada
 
 	return errors.NotFoundError(errors.Errorf("the retention policy id %d does not match", p.ID))
 }
+
+func (r *retentionAPI) requireExecutionInProject(ctx context.Context, p *policy.Metadata, executionID int64) error {
+	exec, err := r.retentionCtl.GetRetentionExec(ctx, executionID)
+	if err != nil {
+		return err
+	}
+	if exec == nil {
+		return errors.New(nil).WithMessage("project: %d, execution id %d not found", p.Scope.Reference, executionID).WithCode(errors.NotFoundCode)
+	}
+	if exec.PolicyID != p.ID {
```

Comment:
except check the policyID, we should check the vendor_type too. the GetRetentionExec doesn't check vendor_type